### PR TITLE
Change matrix function to row major, add tests for them

### PR DESCRIPTION
**Pull request by tonis2**
Overview what this pull request contains

* Change some matrix functions to use row-major layout, cause previously (cause my own mistake) some of them were in column-major

* New `roundf` macro, it's very useful when writing tests for matrixes for example, cause all the values have lots of decimal places.

* New `radians` macro

* Add identity constant for Quaternion

### CODE REVIEW

File: lib/std/math/math.c3 [hunk 1]
```diff
@@ -97,8 +97,6 @@ def complex_identity = complex::identity(<double>);
 
 def Quaternionf = Quaternion(<float>);
 def Quaternion = Quaternion(<double>);
-def quaternionf_identity = quaternion::identity(<float>);
```
Comment (lerno) [inline lib/std/math/math.c3:100]: This should not be removed until 0.6.0. At most add a `@deprecated`

File: lib/std/math/math.c3 [hunk 2]
```diff
@@ -120,6 +118,14 @@ def MATRIX3F_IDENTITY = matrix::IDENTITY3(<float>);
 def MATRIX4_IDENTITY = matrix::IDENTITY4(<double>);
 def MATRIX4F_IDENTITY = matrix::IDENTITY4(<float>);
 
+
+/**
+ * @require types::is_numerical($typeof(x)) `The input must be a numerical value or numerical vector`
+ **/
+macro radians(x) {
+    return x * PI / 180;
```
Comment (lerno) [inline lib/std/math/math.c3:126]: `deg_to_rad` as name perhaps?

File: lib/std/math/math_matrix.c3 [hunk 3]
```diff
@@ -9,7 +9,7 @@ struct Matrix2x2
 			Real m00, m01;
 			Real m10, m11;
 		}
-		Real[4] m;
+		Real[<4>] m;
```
Comment (lerno) [inline lib/std/math/math_matrix.c3:12]: Note that this changes the alignment of the matrix. This might be ok, but it should be mentioned.

Comment (tonis2) [reply lib/std/math/math_matrix.c3:12]: Glad you mentioned, tested it and my vulkan renderer started giving below error cause of this change

```c
Fatal glibc error: malloc assertion failure in sysmalloc: (old_top == initial_top (av) && old_size == 0) || ((unsigned long) (old_size) >= MINSIZE && prev_inuse (old_top) && ((unsigned long) old_end & (pagesize - 1)) == 0)
```

Only reason I made this change was, so I could use `math::round_to_decimals(matrix.m, 4)`
I guess, I'll have to use good old for loop, to make the rounding.

Comment (lerno) [reply lib/std/math/math_matrix.c3:12]: You can do a cast: `math::round_to_decimals((Real[<3>])matrix.m, 4)`

**Comment by lerno** [comment]
Can you add the same tests but for the float variants? So there aren't some problems with conversions with that one.

**Comment by tonis2** [comment]
@lerno 
Do you mean adding tests like below, so both double and float are tested ?

```c
assert(math::round_to_decimals((double[<16>])result.m, 4) == math::round_to_decimals((double[<16>])perspective.m, 4));

assert(math::round_to_decimals((float[<16>])result.m, 4) == math::round_to_decimals((float[<16>])perspective.m, 4));
```

**Comment by data-man** [comment]
What if instead of adding a new function, add a new parameter to `math::ceil/floor/round/etc`?

**Comment by lerno** [comment]
Yes, but I think especially you want to test both Matrix4 and Matrix4f etc.

**Comment by lerno** [comment]
Can you rebase and squash this into a single commit?

**Pull request closed without merging**